### PR TITLE
Attempt to read the token from RESIN_TOKEN env var

### PR DESCRIPTION
**Pull request by jviotti**
- The environment variable has precedence over a saved token.

### CODE REVIEW

File: lib/token.coffee [hunk 1]
```diff
@@ -118,6 +122,7 @@ exports.has = ->
 # token.remove()
 ###
 exports.remove = ->
+	delete process.env.RESIN_TOKEN
```
Comment (dfunckt) [inline lib/token.coffee:125]: Not sure about whether removing the env var is expected. I'd say we don't remove it, but I won't insist.

File: tests/token.spec.coffee [hunk 2]
```diff
@@ -132,6 +172,39 @@ describe 'Token:', ->
...
+						token.set('1234asdf')
+					.nodeify(done)
+
+				afterEach ->
+					delete process.env.RESIN_TOKEN
+
+				it 'should remove the token', (done) ->
+					m.chai.expect(token.has()).to.eventually.be.true
+					token.remove().then ->
+						m.chai.expect(token.has()).to.eventually.be.false
```
Comment (pcarranzav) [inline tests/token.spec.coffee:205]: Shouldn't this be like this for what chai as promised expects?:

```
m.chai.expect(token.has()).to.eventually.be.false.notify(done)
```

File: lib/token.coffee (hunk 1 above)
Comment (petrosagg) [inline lib/token.coffee:125]: I agree. In general `process.env` should be considered const. If you need to keep mutable state about some token you should initialise something else from the environment and then do any mutations there.

**Comment by jviotti** [comment]
Closing for now since the token gets invalidated after a certain time, therefore hardcoding a `RESIN_TOKEN` variable will stop working after a while (and we cannot updated the value of the environment variable outside the application process reliably). This needs discussion.

**Pull request closed without merging**